src/test/java/com/example/onlinebookstore/controller/ShoppingCartControllerTest.java

```diff
@@ -0,0 +1,232 @@
+                    new ClassPathResource("database/books/insert-user.sql")
+            );
+            ScriptUtils.executeSqlScript(
+                    connection,
+                    new ClassPathResource("database/books/insert-shopping-cart.sql")
+            );
+            ScriptUtils.executeSqlScript(
+                    connection,
+                    new ClassPathResource("database/books/insert-cart-items.sql")
+            );
+        }
+    }
+
+    @AfterAll
+    static void afterAll(
+            @Autowired DataSource dataSource
+    ) {
+        teardown(dataSource);
+    }
+
+    @SneakyThrows
+    static void teardown(DataSource dataSource) {
+        try (Connection connection = dataSource.getConnection()) {
+            connection.setAutoCommit(true);
+            ScriptUtils.executeSqlScript(
+                    connection,
+                    new ClassPathResource("database/books/delete-book-category.sql")
+            );
+            ScriptUtils.executeSqlScript(
+                    connection,
+                    new ClassPathResource("database/books/remove-all-books.sql")
+            );
+            ScriptUtils.executeSqlScript(
+                    connection,
+                    new ClassPathResource("database/books/delete-all-category.sql")
+            );
+            ScriptUtils.executeSqlScript(
+                    connection,
+                    new ClassPathResource("database/books/remove-shopping-carts.sql")
+            );
+            ScriptUtils.executeSqlScript(
+                    connection,
+                    new ClassPathResource("database/books/remove-cart-items.sql")
+            );
+            ScriptUtils.executeSqlScript(
+                    connection,
+                    new ClassPathResource("database/books/remove-user.sql")
+            );
+        }
+    }
+
+    @DisplayName("Add book to shopping cart should return success")
+    @WithUserDetails("user1@example.com")
+    @Test
+    void addBookToShopCart_ValidShoppingCartDto_ShouldReturnSuccess() throws Exception {
+        // Given
+        CartItemDto cartItemDto = new CartItemDto();
+        cartItemDto.setBookId(2L);
+        cartItemDto.setQuantity(1);
+        ShoppingCartDto expected = new ShoppingCartDto(1L, 3L,
+                Set.of(new CartItemResponseDto(1L, 1L, "Book1", 1),
+                        new CartItemResponseDto(2L, 2L, "Book2", 1)));
+
+        String jsonRequest = objectMapper.writeValueAsString(cartItemDto);
+
+        // When
+        MvcResult result = mockMvc.perform(post("/cart")
+                .content(jsonRequest)
+                .contentType(MediaType.APPLICATION_JSON)
+        )
+                .andExpect(status().isCreated())
+                .andReturn();
+
+        // Then
+        ShoppingCartDto actual = objectMapper.readValue(result.getResponse()
+                .getContentAsString(), ShoppingCartDto.class);
+
+        Assertions.assertEquals(expected, actual);
+    }
```

Comment:
Please also check by means of repository that DB state was changed as expected (cartItem was created in DB)